OsmAnd/src/net/osmand/plus/inapp/util/BillingManager.java

```diff
@@ -135,18 +135,26 @@ public void onPurchasesUpdated(BillingResult billingResult, @Nullable List<Purch
 	 * Start a purchase flow
 	 */
 	public void initiatePurchaseFlow(final Activity activity, final SkuDetails skuDetails) {
-		initiatePurchaseFlow(activity, skuDetails, null);
+		initiatePurchaseFlow(activity, skuDetails, null, null);
 	}
 
 	/**
 	 * Start a purchase or subscription replace flow
 	 */
-	public void initiatePurchaseFlow(final Activity activity, final SkuDetails skuDetails, final String oldSku) {
+	public void initiatePurchaseFlow(final Activity activity, final SkuDetails skuDetails, final String oldSku, final String purchaseToken) {
 		Runnable purchaseFlowRequest = new Runnable() {
 			@Override
 			public void run() {
-				LOG.debug("Launching in-app purchase flow. Replace old SKU? " + (oldSku != null));
-				BillingFlowParams purchaseParams = BillingFlowParams.newBuilder().setSkuDetails(skuDetails).setOldSku(oldSku).build();
+				LOG.debug("Launching in-app purchase flow. Replace old SKU? " + (oldSku != null && purchaseToken != null));
+				OsmandSettings settings = getSettings(activity);
+				BillingFlowParams.Builder paramsBuilder = BillingFlowParams.newBuilder()
+						.setSkuDetails(skuDetails)
+						.setObfuscatedAccountId(settings.BILLING_USER_ID.get())
```

Comment:
Do not use OsmAnd code in this class. Introduce mObfuscatedAccountId and mObfuscatedProfileId instead of mPayload.